src/evmTest/scala/io/iohk/ethereum/vm/utils/EvmTestEnv.scala

```diff
@@ -83,55 +80,36 @@ trait EvmTestEnv {
     case other => throw new RuntimeException("Invalid call argument")
   }
 
-  class Contract(val name: String, val address: Address) extends Dynamic {
-
-    def applyDynamic(methodName: String)(args: Any*): ContractMethodCall = {
-      callMethod(methodName)(args: _*)
-    }
+  class MockContract(val name: String, val address: Address) extends Dynamic {
 
-    def callMethod(methodName: String)(args: Any*): ContractMethodCall = {
-      if (methodName.contains("("))
-        callWithSignature(methodName)(args: _*)
-      else
-        callWithMethodName(methodName)(args: _*)
-    }
+    def contract: Contract[MockWorldState, MockStorage] =
+      Contract[MockWorldState, MockStorage](address, MockVmInput.blockHeader, internalWorld, contractsAbis(name), config)
 
-    private def callWithMethodName(methodName: String)(args: Any*): ContractMethodCall = {
-      val matchedAbis = contractsAbis(name)
-        .filter { a => a.inputs.size == args.size && a.name == methodName && a.`type` == "function" }
-
-      if (matchedAbis.isEmpty)
-        throw new RuntimeException("No matching ABI found. Please specify the signature")
-      else if (matchedAbis.size > 1)
-        throw new RuntimeException("More than one matching ABI found. Please specify the signature")
-      else {
-        val abi = matchedAbis.head
-        callWithSignature(s"$methodName(${abi.inputs.map(_.`type`).mkString(",")})")(args: _*)
-      }
+    def applyDynamic(methodName: String)(args: Any*): MockContractMethodCall = {
+      val methodCall = contract.applyDynamic(methodName)(args: _*)
+      new MockContractMethodCall(this, methodCall)
     }
 
-    private def callWithSignature(signature: String)(args: Any*): ContractMethodCall = {
-      val signatureHash = ByteString(crypto.kec256(signature.getBytes)).take(4)
-      val callData = args.map(parseArg).foldLeft(signatureHash)(_ ++ _)
-      new ContractMethodCall(this, callData)
+    def callMethod(methodName: String)(args: Any*): MockContractMethodCall = {
+      val methodCall = contract.callMethod(methodName)(args: _*)
+      new MockContractMethodCall(this, methodCall)
     }
 
     def storage: MockStorage = world.storages(address)
   }
 
-  class ContractMethodCall(contract: Contract, callData: ByteString) {
+  class MockContractMethodCall(mockContrack: MockContract,
```

Comment:
Minor detail: Maybe we should remove the `mockContract` parameter as it's not being used?

Reply:
sure!